atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/QosClientBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api;
+
+import java.util.Optional;
+
+import org.immutables.value.Value;
+
+import com.palantir.remoting.api.config.service.ServiceConfiguration;
+
+@Value.Immutable
+public abstract class QosClientBuilder {
+    public abstract Optional<ServiceConfiguration> qosServiceConfiguration();
+
+    @Value.Default
+    public String qosUserAgent() {
```

Comment:
this should be the same user agent we use for timelock (the one passed to the txn manager builder)